src/app/dashboard-widgets/add-codebase-widget/codebase-item/codebase-item.component.spec.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  Component,
+  Input
+} from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+import { MockFeatureToggleComponent } from 'testing/mock-feature-toggle.component';
+import {
+  initContext,
+  TestContext
+} from 'testing/test-context';
+
+import { CodebaseItemComponent } from './codebase-item.component';
+
+import { Codebase } from '../../../space/create/codebases/services/codebase';
+
+@Component({
+  template: `
+    <fabric8-add-codebase-widget-codebase-item [codebase]="codebase"></fabric8-add-codebase-widget-codebase-item>
+  `
+})
+class HostComponent {
+  codebase: Codebase = {
+    attributes: {
+      last_used_workspace: 'foo-workspace',
+      stackId: 'foo-stack'
+    },
+    type: 'mock-codebase'
+  };
+}
+
+@Component({
+  selector: 'codebases-item-workspaces',
+  template: ''
+})
+class MockCodebasesItemWorkspacesComponent {
+  @Input() codebase: Codebase;
+}
+
+describe('AddCodebaseWidget CodebaseItemComponent', () => {
+
+  type TestingContext = TestContext<CodebaseItemComponent, HostComponent>;
+  initContext(CodebaseItemComponent, HostComponent, {
+    declarations: [
+      MockCodebasesItemWorkspacesComponent,
+      MockFeatureToggleComponent
+    ]
+  });
+
+  it('should be instantiable', function(this: TestingContext): void {
+    expect(this.testedDirective).toBeDefined();
```

Comment:
I'd be inclined not to include this.  Thoughts?

Reply:
Sure, I can take it out. I usually write those sorts of tests first just as a sanity check for the spec suite itself before continuing on to other more meaningful tests.